src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -101,21 +120,27 @@ public static string GetExtension(string path)
             if (path == null)
                 return null;
 
+            return new string(path.AsReadOnlySpan());
```

Comment:
@ahsonkhan why isn't there a string constructor that takes a `Span<char>`? This `AsReadOnlySpan()` does not seem to add value here.

Reply:
This code is not a good case study because of it just returns the original `path` in roundabout way. Once it is fixed to call `GetExtension(ReadOnlySpan<char> path)`, the implicit conversion will kick in as expected.

> why isn't there a string constructor that takes a `Span<char>`

There is a string constructor that takes `ReadOnlySpan<char>`. We have a number of cases where we have both Span and ReadOnlySpan method overloads for different reasons, but we have to stop somewhere. It would be API bloat to have both overloads everywhere. There was not a good enough reason to string constructor that takes `Span<char>` yet.